### DISCUSSION TITLE
Why do bagging in good and high quality presets if they refit full afterwards?

### DISCUSSION THREAD

**Discussion by markgoodhead** [autogluon/autogluon#1806] [Q&A] [answered]
I've tried going through the docs to find an answer but I'm confused as to why autogluon does bagging in good and high quality presets if they refit full afterwards?  Isn't it just wasted computation if the bagged models are thrown away with the full one used in their place anyway?

**Comment by Innixma** [comment, marked as answer]
1. Bagging allows us to early stop in the val data in each fold. Refit will use the epoch/iteration found on average in bagged folds, can't be done without first doing the bagged folds.
2. Bagging is required to get out-of-fold predictions for stacking/weighted ensembling. Without out-of-fold predictions / validation predictions, it is impossible to create a smart ensemble.
